modules/services/hercules-ci-agent/default.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.hercules-ci-agent;
+  user = config.users.users.hercules-ci-agent;
+in
+{
+  imports = [ ./common.nix ];
+
+  options.services.hercules-ci-agent = {
+
+    logFile = mkOption {
+      type = types.path;
+      default = "/var/log/hercules-ci-agent.log";
+      description = "Stdout and sterr of hercules-ci-agent process.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    launchd.daemons.hercules-ci-agent = {
+      script = "exec ${cfg.package}/bin/hercules-ci-agent --config ${cfg.tomlFile}";
+
+      path = [ config.nix.package ];
+      environment = {
+        NIX_SSL_CERT_FILE = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
+      };
+
+      serviceConfig.KeepAlive = true;
+      serviceConfig.RunAtLoad = true;
+      serviceConfig.StandardErrorPath = cfg.logFile;
+      serviceConfig.StandardOutPath = cfg.logFile;
+      serviceConfig.GroupName = "hercules-ci-agent";
+      serviceConfig.UserName = "hercules-ci-agent";
+      serviceConfig.WorkingDirectory = user.home;
+      serviceConfig.WatchPaths = [
+        cfg.settings.staticSecretsDirectory
+      ];
+    };
+
+    system.activationScripts.preActivation.text = ''
+      touch '${cfg.logFile}'
+      chown ${toString user.uid}:${toString user.gid} '${cfg.logFile}'
+    '';
+
+    # Trusted user allows simplified configuration and better performance
+    # when operating in a cluster.
+    nix.trustedUsers = [ "hercules-ci-agent" ];
+    services.hercules-ci-agent.settings.nixUserIsTrusted = true;
+
+    users.knownGroups = [ "hercules-ci-agent" ];
+    users.knownUsers = [ "hercules-ci-agent" ];
+
+    users.users.hercules-ci-agent = {
+      uid = mkDefault 399;
+      gid = mkDefault config.users.groups.hercules-ci-agent.gid;
+      home = mkDefault cfg.settings.baseDirectory;
+      name = "_hercules-ci-agent";
```

Comment:
This seems to have caused the removal of the old `hercules-ci-agent` user but not the creation of the `_hercules-ci-agent` user.

Reply:
users/default.nix is confusing attribute keys and names `ConfigName` and `Name` below.

This breaks `users.users.<name>.name`. (`<name>` corresponds to `ConfigName` and `name` corresponds to `Name` below)

```
  # isCreated : [ConfigName] -> ConfigName -> Bool
  isCreated = listOfKnown: name: elem name listOfKnown;
  # isDeleted : Map *ConfigName *Config -> Name -> Bool
  isDeleted = attrs: name: ! elem name (mapAttrsToList (n: v: v.name) attrs);

  # gids : Map GIDString Name
  gids = mapAttrsToList (n: toGID) (filterAttrs (n: v: isCreated cfg.knownGroups /* expecting ConfigName but actual is Name */ v.name) cfg.groups);
  # uids : Map UIDString Name
  uids = mapAttrsToList (n: toUID) (filterAttrs (n: v: isCreated cfg.knownUsers /* expecting ConfigName but actual is Name */ v.name) cfg.users);

  # createdGroups : [GroupConfig]
  createdGroups = mapAttrsToList (n: v: cfg.groups."${v}") cfg.gids;
  # createdGroups : [UserConfig]
  createdUsers = mapAttrsToList (n: v: cfg.users."${v}") cfg.uids;
  # deletedGroups : [GroupConfigName]
  deletedGroups = filter (n: isDeleted cfg.groups n) cfg.knownGroups;
  # deletedUsers : [UserConfigName]
  deletedUsers = filter (n: isDeleted cfg.users n) cfg.knownUsers;
```

Reply:
I guess I was hoping this could support renames. I'll stick with delete and create for the migration.

Reply:
I've turned this finding into an issue https://github.com/LnL7/nix-darwin/issues/294

Reply:
This PR does not depend on a fix for it.

Reply:
Done.